Sources/PayPalWebPayments/PayPalWebCheckoutClient.swift

```diff
@@ -104,6 +104,10 @@ public class PayPalWebCheckoutClient: NSObject {
     /// - Returns: PayPalVaultResult
     /// - Throws: PayPalSDK error if vaulting could not complete successfully
     public func vault(url: URL) {
```

Comment:
Outside the scope of this PR - but why doesn't `PayPalWebCheckoutClient.vault()` accept a request object? The entry point to all of our other SDK features use the request object pattern (see [CardCllient.vault()](https://github.com/paypal/paypal-ios/blob/15fa228521118f64ba9685f496d5c44c51631fb6/Sources/CardPayments/CardClient.swift#L46), [PayPalWebClient.start()](https://github.com/paypal/paypal-ios/blob/15fa228521118f64ba9685f496d5c44c51631fb6/Sources/PayPalWebPayments/PayPalWebCheckoutClient.swift#L35), [PayPalNativeCheckoutClient.start()](https://github.com/paypal/paypal-ios/blob/15fa228521118f64ba9685f496d5c44c51631fb6/Sources/PayPalNativePayments/PayPalNativeCheckoutClient.swift#L43)).

The request object pattern helps us easily add request params as needed as features scale versus needing to add new overload methods.

Reply:
IMO a `PayPalVaultRequest` with a required `setupTokenID` & `url` would help future proof our analytics (in case the URL changes the `approval_session_id` param name), as well as align us better to the Card vaulting feature's public API.

Reply:
I agree we've never discussed this as a group. A `PayPalVaultRequest` object with a `url` property would be consistent with other methods. I'm personally 50 / 50 on adding a `setupTokenID` since it's an extra property, but if it's critical to analytics we should add it in.

Reply:
Captured in DTPPCPSDK-1985.